### PR TITLE
Add correlationId to Context.Operation.Id

### PR DESCRIPTION
**Pull request by SLdragon**

### CODE REVIEW

File: e2e-diagnostics/run.csx [hunk 1]
```diff
@@ -109,6 +114,8 @@ public class AI
         telemetry.Context.Cloud.RoleName = endpointName;
         telemetry.Context.Cloud.RoleInstance = DefaultRoleInstance;
         telemetry.Context.Operation.ParentId = dependencyId;
+        //This telemtry is used to draw the application Map, so we do not save its correlation id
```
Comment (SergeyKanzhelev) [inline e2e-diagnostics/run.csx]: I don't understand this comment

Comment (SLdragon) [reply e2e-diagnostics/run.csx]: Recently our e2e diagnostics only have 3 types of logs generated by the Azure IoT Hub: D2C log, Ingress log and Egress log. We do not have logs which record the latency processed by the endpoint, and we send this request to AI to draw the endpoint instance in the Application Map, it doesn't contains any useful information. In the future, if the endpoint support e2e diagnostics then we can add correlation Id to this log.

| # | type  | Operation parent | target  | Cloud role |
|---|---|---|---|---|
| D2C  | dependency  | / | IoT Hub|Device |
| Ingress | request | D2C | / | IoT Hub| 
| Egress | dependency | Ingress | Endpoint | IoT Hub| 
| * | request | Egress | / | Endpoint|  

clipboard

Comment (SLdragon) [reply e2e-diagnostics/run.csx]: Hi, @SergeyKanzhelev, recently i update the code to add properties and name to the properties, please help review the code, thanks a lot!

File: e2e-diagnostics/run.csx [hunk 2]
```diff
@@ -79,8 +92,13 @@ public class AI
             Duration = new TimeSpan(0, 0, 0, 0, egressLatency),
             Target = endpointName,
             Success = !hasError,
+            Name = "Egress Latency"
         };
 
+        dependencyTelemetry.Properties["parentSpanId"] = properties.parentSpanId;
```
Comment (SergeyKanzhelev) [inline e2e-diagnostics/run.csx]: should the `dependencyTelemetry.Context.Operation.ParentId` be used instead?

File: e2e-diagnostics/run.csx [hunk 3]
```diff
@@ -79,8 +92,13 @@ public class AI
             Duration = new TimeSpan(0, 0, 0, 0, egressLatency),
             Target = endpointName,
             Success = !hasError,
+            Name = "Egress Latency"
         };
 
+        dependencyTelemetry.Properties["parentSpanId"] = properties.parentSpanId;
+        dependencyTelemetry.Properties["endpointName"] = properties.endpointName;
+        dependencyTelemetry.Properties["endpointType"] = properties.endpointType;
```
Comment (SergeyKanzhelev) [inline e2e-diagnostics/run.csx]: depending on semantics of `endpointName` and `endpointType` consider putthing one of them to `dependencyTelemetry.Context.Cloud.RoleName`

**Review by SergeyKanzhelev — APPROVED** [review]

File: e2e-diagnostics/run.csx (hunk 2 above)
Comment (SLdragon) [reply e2e-diagnostics/run.csx]: This is parentSpanId, and we do not need to add it to ParentId

File: e2e-diagnostics/run.csx (hunk 3 above)
Comment (SLdragon) [reply e2e-diagnostics/run.csx]: We have already added endpoint name to RoleName

**Pull request merged**